graphql_compiler/schema/__init__.py

```diff
@@ -415,6 +415,17 @@ def _parse_datetime_value(value: Any) -> datetime:
         )
 
 
+def _parse_decimal_value(value: Any) -> Decimal:
+    """Parse decimal value."""
+    if isinstance(value, float):
+        # Decimal(float(1.1)) is Decimal('1.100000000000000088817841970012523233890533447265625'),
+        # which is probably not what the user intended. Inflating the intended level of precision
+        # like this can cause errors with database types that are not configured with precision
+        # that high. Just use Decimal or str values if you're serious about precision.
+        return Decimal(str(value))
```

Comment:
I'm not yet 100% sure on this approach. An alternative is to disallow floats for GraphQLDecimal.